editor/src/tool/tools/path.rs

```diff
@@ -211,6 +223,170 @@ impl Fsm for PathToolFsmState {
+						mouse_proximity: f64,
+					}
+
+					impl Point {
+						fn new(_position: DVec2, point_type: PointType, mouse_proximity: f64) -> Self {
+							Self { point_type, mouse_proximity }
+						}
+					}
+
+					// TODO simplify the following block
+					let select_threshold = 6.;
+					let select_threshold_squared = select_threshold * select_threshold;
+
+					for (shape_offset, shape_to_draw) in shapes_to_draw.iter().enumerate() {
+						let segment = shape_manipulator_points(shape_to_draw);
+
+						for anchor in segment.anchors {
+							let d2 = mouse_pos.distance_squared(anchor);
+							if d2 < select_threshold_squared {
+								points.push(Point::new(
+									anchor,
+									PointType::Anchor {
+										anchor_i,
+										layer_path: shape_to_draw.layer_path.clone(),
+										shape_offset,
+									},
+									d2,
+								));
+							}
+							anchor_i += 1;
+						}
+
+						for (_, handle) in segment.handles.into_iter().enumerate() {
+							let d2 = mouse_pos.distance_squared(handle);
+							if d2 < select_threshold_squared {
+								points.push(Point::new(
+									handle,
+									PointType::Handle {
+										handle_i,
+										layer_path: shape_to_draw.layer_path.clone(),
+										shape_offset,
+									},
+									d2,
+								));
+							}
+							handle_i += 1;
+						}
+					}
+
+					points.sort_by(|a, b| a.mouse_proximity.partial_cmp(&b.mouse_proximity).unwrap_or(std::cmp::Ordering::Equal));
+					let closest_point_within_click_threshold = points.first();
+
+					if let Some(point) = closest_point_within_click_threshold {
+						let path = match point.point_type {
+							PointType::Anchor {
+								anchor_i,
+								ref layer_path,
+								shape_offset,
+							} => {
+								data.selected_shapes = shapes_to_draw;
+								let shape = &data.selected_shapes[shape_offset];
+								let path = shape.path.clone();
+								let bez: Vec<PathEl> = (&path).into_iter().collect();
+								let transformed = shape.transform.inverse().transform_point2(input.mouse.position);
+								data.selection.bez_segment_id = closest_anchor(&bez, Vec2::new(transformed.x, transformed.y));
+								data.selection.bez_path_elements = bez;
+								data.selection.closest_layer_path = layer_path.clone();
+								data.selection.closest_shape_id = shape_offset;
+								data.anchor_marker_pool[anchor_i].clone()
+							}
+							PointType::Handle {
+								handle_i,
+								ref layer_path,
+								shape_offset,
+							} => {
+								// TODO make this work for the handles, right now just selects the anchors
+								data.selected_shapes = shapes_to_draw;
+								let shape = &data.selected_shapes[shape_offset];
+								let path = shape.path.clone();
+								let bez: Vec<PathEl> = (&path).into_iter().collect();
+								let transformed = shape.transform.inverse().transform_point2(input.mouse.position);
+								data.selection.bez_segment_id = closest_anchor(&bez, Vec2::new(transformed.x, transformed.y));
```

Comment:
Use new fn `closest_handle` instead? // Seems like it checks bezier curve endpoints (anchor points) but not its other control points (direction handles)?  Note how, on an ellipse, pressing down on a handle marks it correctly, but then when dragging, it drags the closest anchor instead.

Reply:
I took a very slap dash approach to getting this working, the cloest_anchors function doesn't even consider Handles yet. Overall it does have incorrect behavior with anything regarding handles or the close path invisible point. Would love a better way of finding the indexes without brute forcing etc. 

I was just pushing to get this closer to the finish line for game dev meetup this morning. I would be happy to sit down and work on it today!